File: Backend/src/routes/category.routes.js
import { Router } from 'express';
import { body } from 'express-validator';
import {
  getCategories,
  createCategory,
  updateCategory,
  deleteCategory,
} from '../controllers/category.controller.js';
import { protect } from '../middlewares/auth.middleware.js';
import { validate } from '../middlewares/validate.middleware.js';

const router = Router();

// All routes require authentication
router.use(protect);

router.get('/', getCategories);

router.post(
  '/',
  [body('name').notEmpty().withMessage('Category name is required')],
  validate,
  createCategory
);

router.put(
  '/:id',
  [body('name').optional().notEmpty().withMessage('Category name cannot be empty')],
  validate,
  updateCategory
);

router.delete('/:id', deleteCategory);

export default router;

File: FastAPI_Backend/app/routes/category_routes.py
from fastapi import APIRouter, Depends
from app.models.category import CategoryCreate, CategoryUpdate
from app.controllers import category_controller
from app.middlewares.auth import get_current_user

router = APIRouter(prefix="/categories", tags=["Categories"])


@router.get("/")
async def get_categories(current_user=Depends(get_current_user)):
    return await category_controller.get_categories(str(current_user.id))


@router.post("/", status_code=201)
async def create_category(body: CategoryCreate, current_user=Depends(get_current_user)):
    return await category_controller.create_category(body, str(current_user.id))


@router.put("/{category_id}")
async def update_category(category_id: str, body: CategoryUpdate, current_user=Depends(get_current_user)):
    return await category_controller.update_category(category_id, body, str(current_user.id))


@router.delete("/{category_id}")
async def delete_category(category_id: str, current_user=Depends(get_current_user)):
    return await category_controller.delete_category(category_id, str(current_user.id))
